### PR TITLE
Add descriptive message for empty parameters when using svcat (#2051)

### LINKED ISSUE #2051: Don't show empty parameters
When an instance's parameters are empty, but not nil, svcat is currently printing it out with `{}`. It would be better if we could detect that it's empty and not display the parameters.

**Current Output**
```
$ svcat describe inst redis1
  Name:        redis1
  Namespace:   default
  Status:      Ready - The instance was provisioned successfully @ 2018-05-18 01:22:15 +0000 UTC
  Class:       redis
  Plan:        4-0-9

Parameters:
  {}

Bindings:
No bindings defined
```

**Desired Output**
```
$ svcat describe inst redis1
  Name:        redis1
  Namespace:   default
  Status:      Ready - The instance was provisioned successfully @ 2018-05-18 01:22:15 +0000 UTC
  Class:       redis
  Plan:        4-0-9

Bindings:
No bindings defined
```

The code responsible for writing the parameters is found here:

https://github.com/kubernetes-incubator/service-catalog/blob/30be2f447d5abaa1f24891b2ff226d270bd330ca/cmd/svcat/output/yaml.go#L48-L62

There isn't an existing test for that function, so as part of a pull request, adding a new test file, yaml_test.go, and 2-3 test cases around that function would be required.

### PR DESCRIPTION
**Pull request by artmello**
* Add unit tests to check Parameters output

Closes #2051

### CODE REVIEW

**Comment by n3wscott** [comment]
https://github.com/kubernetes-incubator/service-catalog/issues/2051#issuecomment-390669912

File: cmd/svcat/output/yaml_test.go [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+
+func TestWriteParameters(t *testing.T) {
+	testcases := []struct {
+		name       string                // Test name
+		parameters *runtime.RawExtension // Parameters tested
+		output     string                // Expected output
+	}{
+		{"Nil parameter", nil, ""},
+		{"JSON w/data parameter", &runtime.RawExtension{Raw: []byte(`{"foo":"bar"}`)}, "\nParameters:\n  foo: bar\n"},
+		{"JSON empty parameter", &runtime.RawExtension{Raw: []byte(`{}`)}, "\nParameters:\n  {}\n"},
```
Comment (carolynvs) [inline cmd/svcat/output/yaml_test.go]: Sorry for my miscommunications on this one! :-(

When we said that json output should still print `{}` we were actually referring to a different command, e.g. `svcat get binding NAME -o json`. 

For this testcase ("json empty parameter"), the intent is to print the yaml representation of the parameters or fallback to the raw string, the expected output should also be `"\nParameters:\n  No parameters defined\n"`.

Maybe moving the `if string(parameters.Raw) == ""` check to before the json marshalling, and also testing for `{}` would accomplish that?

**Comment by carolynvs** [comment]
@artmello If you are interested in how to test this manually with the cli to see the final command output, you can run the commands below. It's not necessary for this PR, but will get your a bit more familiar with the cli and help you see where the changes that you are making show up.

1. Setup a cluster with service catalog installed (see http://svc-cat.io/docs/install/) and wait for the pods to be ready.
1. Install the svcat cli (instructions on that same page above)
1. Install Minibroker and wait for the pods to be ready.
    ```
    helm repo add minibroker https://minibroker.blob.core.windows.net/charts
    helm install --name minibroker --namespace minibroker minibroker/minibroker
    ```
1. Provision a service with minibroker
    ```
    svcat provision mysqldb --class mysql --plan 5-7-14
    ```
1. Create a binding for the service
    ```
    svcat bind mysqldb
    ```
1.  Run the command that we are changing with this PR: `svcat describe binding mysqldb`. You should see the empty parameters `{}` printed without your fix and with your fix you'll see `No parameters defined`.

**Comment by carolynvs** [comment]
@artmello Sweet! Let me know if you run into trouble with my steps. Feel free to ping me in the kubernetes slack for a quicker response.

**Comment by carolynvs** [comment]
Kicking CI because of a known test flake (#2036). I'll merge once it passes

**Pull request merged**